### ISSUE TITLE
Refactor/Remove ColumnReorderApi and ColumnResizeApi

### ISSUE THREAD

**Issue by dtassone** [mui-org/material-ui-x#1201]
## Summary 💡

Those methods attached to apiRef are actually event handlers and should not be on apiRef. 
ApiRef goal is to allow users to manipulate the grid.
`startResizeOnMouseDown`, `onColItemDragStart`, `onColHeaderDragOver`... are not meaningful.

Something like `options.onColumnResize` or `options.onColumnReordered` are more appropriate.
To get the effect required by those methods, we can just publish the event from the component and subscribe to the event in the hook to handle it.

We can also provide a `setColumnWidth` method on the api that would allow to only update the col width, using the state, and firing onColumnResize once done.

Similarly we can provide a `setColumnIndex`method on the api that would allow to update the col position, using the state, and firing onColumnReorder event once done.

related to #1167

**Comment by DanailH** [comment]
I agree. There was a similar request for having an `onColumnReordered` on the grid itself.

**Comment by bushev** [comment]
Hey @DanailH, 

The feature is quite essential for our app. What's the approximate ETA?

**Comment by dtassone** [comment]
FYI @m4theushw is going to refactor `onColumnReorder` as part of #1250 

We can create a separate issue for `onColumnResize`
